cde/sdk/go/pkg/cdf/events/extensions/artifact_extension.go

```diff
@@ -0,0 +1,67 @@
+package extensions
+
+import (
+	"github.com/cloudevents/sdk-go/v2/binding"
+	"github.com/cloudevents/sdk-go/v2/types"
+)
+
+const (
+	ArtifactIdExtension      = "artifactid"
+	ArtifactNameExtension    = "artifactname"
+	ArtifactVersionExtension = "artifactversion"
+)
+
+// ArtifactExtension represents the extension for extension context
+type ArtifactExtension struct {
+	ArtifactId      string `json:"artifactid"`
+	ArtifactName    string `json:"artifactname"`
+	ArtifactVersion string `json:"artifactversion"`
+}
```

Comment:
I think we should have a single type that we use to both create cloud events and parse them back. 
`ArtifactExtension` and `ArtifactEventParams` could be the same struct.
We probably could have an interface as well and methods to produce cloud events and parse them.
It should be fine for now to keep moving but we'll need to focus on refactoring once the MVP PoC is done :)

Reply:
@afrittoli yeah.. that sounds good to me.. for some reason I think that we need two structs.. i will try to make it one as soon as we finish the initial PoC.. I am currently blocked by this PR